pkg/validation/node_api_adapter.go

```diff
@@ -46,19 +52,99 @@ type NodeAPIAdapter struct {
 
 	//TODO: convert to arg on WaitForNodeToBe
 	// K8s timeout on method call
-	timeout time.Duration
+	timeout      time.Duration
+	clientConfig clientcmd.ClientConfig
 }
 
-func NewNodeAPIAdapter(client kubernetes.Interface, timeout time.Duration) (*NodeAPIAdapter, error) {
+func NewNodeAPIAdapter(client kubernetes.Interface, timeout time.Duration, clientConfig clientcmd.ClientConfig) (*NodeAPIAdapter, error) {
```

Comment:
May need syntax assistance with this.  Should refactor k8s clients and kops clients to our fakes.